Source/Core/Core/ConfigLoaders/GameConfigLoader.cpp

```diff
@@ -0,0 +1,415 @@
+
+    {{"Video_Stereoscopy", "StereoConvergence"},
+     {Config::System::GFX, "Stereoscopy", "StereoConvergence"}},
+    {{"Video_Stereoscopy", "StereoEFBMonoDepth"},
+     {Config::System::GFX, "Stereoscopy", "StereoEFBMonoDepth"}},
+    {{"Video_Stereoscopy", "StereoDepthPercentage"},
+     {Config::System::GFX, "Stereoscopy", "StereoDepthPercentage"}},
+
+    // UI
+    {{"EmuState", "EmulationStateId"}, {Config::System::UI, "EmuState", "EmulationStateId"}},
+    {{"EmuState", "EmulationIssues"}, {Config::System::UI, "EmuState", "EmulationIssues"}},
+    {{"EmuState", "Title"}, {Config::System::UI, "EmuState", "Title"}},
+};
+
+static ConfigLocation MapINIToRealLocation(const std::string& section, const std::string& key)
+{
+  auto it = ini_to_location.find({section, key});
+  if (it == ini_to_location.end())
+  {
+    // Try again, but this time with an empty key
+    // Certain sections like 'Speedhacks' has keys that are variable
+    it = ini_to_location.find({section, ""});
+    if (it != ini_to_location.end())
+      return it->second;
+
+    // Attempt to load it as a configuration option
+    // It will be in the format of '<System>.<Section>'
+    std::istringstream buffer(section);
+    std::string system_str, config_section;
+
+    bool fail = false;
+    std::getline(buffer, system_str, '.');
+    fail |= buffer.fail();
+    std::getline(buffer, config_section, '.');
+    fail |= buffer.fail();
+
+    if (!fail)
+      return {Config::GetSystemFromName(system_str), config_section, key};
+
+    WARN_LOG(COMMON, "Unknown game INI option in section %s: %s", section.c_str(), key.c_str());
+    return {Config::System::Main, "", ""};
+  }
+
+  return ini_to_location[{section, key}];
+}
+
+static std::pair<std::string, std::string> GetINILocationFromConfig(const ConfigLocation& location)
+{
+  auto it = std::find_if(ini_to_location.begin(), ini_to_location.end(),
+                         [&](const auto& entry) { return entry.second == location; });
+
+  if (it != ini_to_location.end())
+    return it->first;
+
+  // Try again, but this time with an empty key
+  // Certain sections like 'Speedhacks' have keys that are variable
+  it = std::find_if(ini_to_location.begin(), ini_to_location.end(), [&](const auto& entry) {
+    return std::tie(entry.second.system, entry.second.key) ==
+           std::tie(location.system, location.key);
+  });
+  if (it != ini_to_location.end())
+    return it->first;
+
+  WARN_LOG(COMMON, "Unknown option: %s.%s", location.section.c_str(), location.key.c_str());
+  return {"", ""};
+}
+
+// INI Game layer configuration loader
+class INIGameConfigLayerLoader : public Config::ConfigLayerLoader
+{
+public:
+  INIGameConfigLayerLoader(const std::string& id, u16 revision, bool global)
+      : ConfigLayerLoader(global ? Config::LayerType::GlobalGame : Config::LayerType::LocalGame),
+        m_id(id), m_revision(revision)
+  {
+  }
+
+  void Load(Config::Layer* config_layer) override
```

Comment:
This function should likely be separated out into smaller functions that are part of the class. This is kind of massive.